src/cromshell/logs/command.py

```diff
@@ -57,222 +77,281 @@ def main(
+                all_task_logs[call].append(
+                    filter_task_logs_from_workflow_metadata(
+                        scatter["subWorkflowMetadata"],
+                        requested_status=requested_status,
+                    )
                 )
-
-        # If no subworkflow is found then print status summary for the task
         else:
-            did_print = (
-                print_task_logs(
-                    task=task,
-                    indent=indent,
-                    workflow_metadata=workflow_metadata,
-                    status_keys=status_keys,
-                    cat_logs=cat_logs,
-                )
-                or did_print
-            )
+            all_task_logs[call] = []
+            for index in index_list:
+                if (
+                    "ALL" in requested_status
+                    or index.get("executionStatus") in requested_status
+                ):
+                    all_task_logs[call].append(
+                        {
+                            "attempt": index.get("attempt"),
+                            "backendLogs": get_backend_logs(task_instance=index),
+                            "backend": index.get("backend"),
+                            "executionStatus": index.get("executionStatus"),
+                            "shardIndex": index.get("shardIndex"),
+                            "stderr": index.get("stderr"),
+                            "stdout": index.get("stdout"),
+                        },
+                    )
 
-    return did_print
+    check_for_empty_logs(
+        workflow_logs=all_task_logs,
+        workflow_id=workflow_metadata["id"],
+        requested_status=requested_status,
+    )
 
+    return all_task_logs
 
-def print_task_logs(
-    task: str,
-    indent: str,
-    workflow_metadata: dict,
-    status_keys: list,
-    cat_logs: bool,
-) -> bool:
-    """
-    Prints the backend logs from the workflow
-    :param task: Name of the task
-    :param indent: Indent string given as a string of "\t" characters,
-    used to indent print out
-    :param workflow_metadata: Metadata of the workflow to process
-    :param status_keys: Determines what logs to show based on call status
-    :param cat_logs: Will use GCS to attempt to print the logs
-    :return: true if any logs were printed
+
+def print_task_level_logs(all_task_log_metadata: dict, cat_logs: bool) -> None:
+    """Print the logs from the workflow metadata
+    task_logs_metadata: {call_name:[index1{task_log_name: taskvalue}, index2{...}, ...], call_name:[], ...}
+
+    Args:
+        all_task_log_metadata (dict): All task logs metadata from the workflow
+        cat_logs (bool): Whether to print the logs
     """
 
-    did_print = False
+    for call, index_list in all_task_log_metadata.items():
+        print(f"{call}:")
+        for call_index in index_list:
+            if call_index is not None:
+                print_file_like_value_in_dict(
```

Comment:
If we define a `class` like recommended above, this could be a method in that class.